platform/pulpcore/tasking/tasks.py

```diff
@@ -243,14 +245,25 @@ def apply_async(self, tags=[], group_id=None, args=None, kwargs=None, **options)
 
     def __call__(self, *args, **kwargs):
         """
-        Set the :class:`pulpcore.app.models.Task` object in the running state and log some output.
+        Init code for our task.
+
+        Set the :class:`pulpcore.app.models.Task` object in the running state,
+        associate the worker, and log a message.
 
         Skip the status updating if the task is called synchronously.
         """
-        if not self.request.called_directly:
-            task_status = TaskStatus.objects.get(pk=self.request.id)
-            task_status.set_running()
-        _logger.debug("Running task : [%s]" % self.request.id)
+        task_status = TaskStatus.objects.get(pk=self.request.id)
+        task_status.set_running()
+
+        try:
+            worker = Worker.objects.get(name=self.request.hostname)
```

Comment:
We should only update if the worker exists and the `worker` field is None. I think this needs an additional `worker=None` option on this `get()` call.